CMakeLists.txt

```diff
@@ -598,6 +598,17 @@ endif()
 
 message(STATUS "Using Lua interpreter: ${LUA_PRG}")
 
+# Some of the code generation still relies on stable table ordering in order to
+# produce reproducible output - specifically the msgpack'ed data in
+# funcs_metadata.generated.h and ui_events_metadata.generated.h. This should
+# ideally be fixed in the generators, but until then as a workaround you may provide
```

Comment:
If this unblocks you then perhaps it's ok as a temporary workaround, but please leave a note athttps://github.com/neovim/neovim/issues/20124 and the commit message, explaining the steps needed to unwind this temporary workaround. It looks like the steps are : 

1. fix the ordering of msgpack'd data in `funcs_metadata.generated.h` and `ui_events_metadata.generated.h`
2. ?

Reply:
btw, does this _actually_ unblock you (e.g. are you packaging for a distro?), or is this only theoretical currently?

Reply:
AFAICS the ordering of msgpack'd data is the only remaining work that requires this workaround. Improved the commit message, happy to also add it to #20124 when this part is merged.

> btw, does this _actually_ unblock you (e.g. are you packaging for a distro?), or is this only theoretical currently?

Yes, this helps neovim be reproducible for nixos (https://github.com/NixOS/nixpkgs/pull/208103). Of course we could also keep the patch locally in nixpkgs, but I prefer upstreaming as much as possible - so perhaps other distro's can also leverage it.

Reply:
> Of course we could also keep the patch locally in nixpkgs, but I prefer upstreaming as much as possible

absolutely, we are always interested in avoiding unnecessary work for package maintainers.